meson.build

```diff
@@ -3,13 +3,38 @@ project('abseil-cpp', 'cpp',
   license : 'Apache-2.0',
   default_options : [
     'cpp_std=c++11',
+    'warning_level=0'
   ])
 
 cpp = meson.get_compiler('cpp')
 
 cpp_flags = [
-  '-Wno-sign-compare'
+  '-Wcast-qual',
+  '-Wconversion-null',
+  '-Wformat',
+  '-Wformat-security',
+  '-Wmissing-declarations',
+  '-Woverlength-strings',
+  '-Wpointer-arith',
+  '-Wundef',
+  '-Wunused-local-typedefs',
+  '-Wunused-result',
+  '-Wvarargs',
+  '-Wvla',
+  '-Wwrite-strings',
+  '-DNOMINMAX',
 ]
+
+if build_machine.cpu_family() in ['x86_64', 'amd64', 'AMD64']
+  cpp_flags += [ '-maes', '-msse4.1' ]
+elif build_machine.cpu_family().to_lower().contains('arm') or build_machine.cpu_family() == 'aarch64'
```

Comment:
Also these checks seem strange.

Reply:
I'm following the cmake rules from here:
https://github.com/abseil/abseil-cpp/blob/master/absl/copts/AbseilConfigureCopts.cmake#L21.

Reply:
They must have very special ARM processors at Google.